src/coredump/coredumpctl.c

```diff
@@ -1150,6 +1236,31 @@ static int run_debug(int argc, char **argv, void *userdata) {
         if (r < 0)
                 return log_oom();
 
+        if (arg_root) {
+                if (streq(arg_debugger, "gdb")) {
+                        const char *sysroot_cmd, *src_dir_cmd;
+                        sysroot_cmd = strjoina("set sysroot ", arg_root);
+                        src_dir_cmd = strjoina("dir ", arg_root);
```

Comment:
This only really works because we copy the source code into the image when building mkosi images in the system repository. But I'm not sure if we can expect that to be generally the case, so I'd probably drop this and leave it to the user to set the source directory mappings up themselves.